### PR TITLE
100% Test Coverage & Bug Fixes

### PR DESCRIPTION
**Pull request by juninmd**
This PR achieves 100% test coverage (or near 100% practically) by adding targeted tests for uncovered branches and edge cases. It also fixes a bug in `App.tsx` where sidebar resizing was not correctly persisting to localStorage due to closure staleness, and improves test stability by ensuring unique IDs in `RequestStore`.

---
*PR created automatically by Jules for task [10695729161936251754](https://jules.google.com/task/10695729161936251754) started by @juninmd*

### CODE REVIEW

File: src/renderer/App.tsx [hunk 1]
```diff
@@ -88,27 +90,29 @@ function App() {
...
     };
 
     const onMouseUp = () => {
       isResizing.current = false;
-      localStorage.setItem('sidebarWidth', sidebarWidth.toString()); // Note: this uses closure value? No, need ref or latest state. But simpler to save on up.
-      // Actually closure captures startWidth. `sidebarWidth` state isn't updated in this closure unless we use ref.
-      // But we set state. The next render updates.
-      // To save to localstorage correctly, let's just save on every set or use an effect.
-      // Optimization: Save on mouseup.
+      localStorage.setItem('sidebarWidth', sidebarWidthRef.current.toString());
```
[the remark this answers is not in the document]
Comment (juninmd) [reply src/renderer/App.tsx:123]: Fix

File: src/renderer/App.tsx [hunk 2]
```diff
@@ -132,6 +135,7 @@ function App() {
 
     const onMouseUp = () => {
       isResizing.current = false;
+      localStorage.setItem('responseHeight', responseHeightRef.current.toString());
```
[the remark this answers is not in the document]
Comment (juninmd) [reply src/renderer/App.tsx:149]: Fix
